cachi2/core/package_managers/bundler/parser.py

```diff
@@ -64,6 +66,33 @@ class GitDependency(_GemMetadata):
     url: AcceptedUrl
     ref: AcceptedGitRef
 
+    @property
+    def repo_name(self) -> str:
+        """Extract the repository name from the URL."""
+        parse_result = urlparse(str(self.url))
+        path = Path(parse_result.path.removesuffix(".git"))
+        return path.name
+
+    def download_to(self, output_dir: RootedPath) -> None:
+        """Download git repository to the output directory with a specific name."""
+        short_ref_length = 12
+        short_ref = self.ref[:short_ref_length]
+
+        git_repo_path = output_dir.join_within_root(f"{self.repo_name}-{short_ref}")
+        if git_repo_path.path.exists():
```

Comment:
Does this logic belong here? Won't components be eventually deduplicated elsewhere?

Reply:
It may happen that the Gemfile.lock contains various duplicate or transitive dependencies, and then `mkdir(parents=True)` will raise `FileExistsError`.

@brunoapimentel sent me this repo https://github.com/brunoapimentel/chatwoot/tree/hermetic-build-test
as an example while "collaborating" on the design document.

Reply:
Thank you for confirming this!